ChangeLog

```diff
@@ -62,6 +62,9 @@ Bugfix
    * Fix issue in RSA key generation program programs/x509/rsa_genkey
      where the failure of CTR DRBG initialization lead to freeing an
      RSA context without proper initialization beforehand.
+   * Fix bug in cipher decryption with MBEDTLS_PADDING_ONE_AND_ZEROS that
+     sometimes accepted invalid padding. (Not used in TLS.) Found and fixed
+     by Micha Kraus.
```

Comment:
@hanno-arm @Patater I forced-push a replacement for the last commit in the PR, with the changelog entry tweaked per the discussion at https://github.com/ARMmbed/mbedtls/pull/1279#discussion_r162208686 .